### DISCUSSION TITLE
Feature Request- Discover Filtering and sorting

### DISCUSSION THREAD

**Discussion by Smensink** [sct/overseerr#1872] [Feature Requests]
It would be great to filter items in the discovery (Trending/popular/etc) tab by: release year, availability, TV/Movie, Genre etc.
The ability to sort by rating or budget would also be useful

**Comment by capu57** [comment]
By rating do you mean MPAA rating or what ever rating system is used where you are.  Or do you mean critique rating?

I personally would like to be able to see a MPAA style rating filter.  So when I am searching actions movies I can limit to Rated "R".  Or if looking for something for my kids to watch I could limit it to G or PG

**Reply by nosajthenitram** [reply]
I've started working on a patch that will pull the allowed ratings from Plex and then apply them as a filter to the content that is displayed in the discover pages. Taking me some time to learn how Overseer works, but making some decent progress.

**Reply by i-am-at0m** [reply]
> I've started working on a patch that will pull the allowed ratings from Plex and then apply them as a filter to the content that is displayed in the discover pages. Taking me some time to learn how Overseer works, but making some decent progress. 

How's it going?

**Comment by tobbbles** [comment]
When in a Discover category (e.g. Trending) I'd expect to see the following filters and sorting options:

Sort by:
- Latest (by release date)
- Rating (IMDB, or Rotten Tomatoes)

Filter:
- Category (All options enabled by default, selecting a category de-selects everything and enables just the selection, ability to multi-select, with a "select all" and "deselect all" button

for me, these three alone would be enough to get going. I foresee the biggest point of contention being what constitutes a rating; potentially opening a configuration setting to assign the 'rating source' to IMDB, RT, or another.

**Reply by TheIndieArmy** [reply]
I agree. I am still surprised you're unable to sort series by latest episode release, and only by first air date. I want to be able to go to Overseer, set the filter to Netflix, and see everything that released in the last week. Not just the brand-new series that released.

**Comment by Xaositek** [comment]
While I'm 2 years late to the party, I feel like any overhaul to sorting should carry over to the Requests tab also. I manually manage the Requests queue, so being able to sort by release dates (or heck even have them visible on the Requests screen) would save me from opening each request to see if the item is available.

**Comment by HerdOn304** [comment]
This feature request would be fantastic! Thanks to all who work on this project!

**Comment by MalronWall** [comment]
Hey! 😄
This is a feature that my users (family mainly and me) have wanted to see for several years now, it would be great!
Do you know what's happened since this 2021 request was created? 🙏🏻

**Comment by HerdOn304** [comment]
I haven't a clue what the status is. I would love it as well. If you know
how to get it on their radar as an enhancement I'm all ears.

**Comment by jasontucker** [comment]
I'd love to be able to sort the "Appearances" screen on an actor a few different ways, release date would be nice. Other places where a grid of posters is being displayed having a sort feature would be very nice.

**Comment by HerdOn304** [comment]
This is exactly my request as well. They have the premier dates seemingly
written to a table. I'm no webdev, but I don't think it's that hard to
create a filter based on that.

**Comment by drohack** [comment]
Please add a filter by availability!
For movies specifically you already have the "Status" information like "In Production", "Post Production", "Released".
But it would be even more awesome if you pulled in the Radarr "Status" which keeps movies as "Not Available" till the DVD release. 

This way I can hide all of the movies that it's not able to grab yet, and only focus on what's actually available to grab/search for.

This could be extended to TV Shows that we have all of the available episodes grabbed, it could hide these shows that the future episodes are not Available yet.

**Comment by maxgoldberg** [comment]
Just sorting by date would be so nice.

**Comment by HerdOn304** [comment]
Who is my community webdev that can help me figure out this sorting thing? I wish I was stronger with web stuff, but I might be able to help with the logic, or poke around a DB to figure out what fields we can key off of.
